Jenkinsfile

```diff
@@ -213,8 +213,10 @@ lock(resource: "build-${params.STREAM}") {
                     // check if there's a previous release we should use as parent
                     if (releases["releases"].size() > 0) {
                         def commit_obj = releases["releases"][-1]["commits"].find{ commit -> commit["architecture"] == basearch }
-                        parent_commit = commit_obj["checksum"]
-                        parent_version = releases["releases"][-1]["version"]
+                        if (commit_obj != null) {
```

Comment:
I believe I hit this. Is it possible that there's an existing release that only contains info for one architecture? so I'm thinkink we hit this if we are doing a first run for a new architecture

Reply:
Yup you're right, forgot the schema for this.

But now this raises an interesting point. Let's say we do a release A with arch1 and arch2 in it. Then do a release B with arch1 and without arch2 because it's broken and we don't want to block. When we do a release C, we want the OSTree parent on arch1 to be B, but on arch2 to be A.

So I think what we actually want here is to keep going backwards in the release history until we hit the architecture that we want. (Or obviously if none are found, then assume there is no parent.)

Reply:
Something like:

```groovy
for (release in releases["releases"].reverse()) {
    def commit_obj = release["commits"].find{ commit -> commit["architecture"] == basearch }
    if (commit_obj != null) {
        parent_commit = commit_obj["checksum"]
        parent_version = release["version"]
    }
}
```

There's probably a more functional way to do this.

Reply:
ok added a commit for this